sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/MiscBenchmark.scala

```diff
@@ -130,6 +130,18 @@ object MiscBenchmark extends SqlBasedBenchmark {
         df.selectExpr("*", "explode(arr) as arr_col")
           .select("col", "arr_col.*").count
       }
+
+      codegenBenchmark("generate big nested struct array", M) {
+        import spark.implicits._
+        val df = spark.sparkContext.parallelize(Seq(("1",
+          Array.fill(M)({
+            val i = math.random
+            (i.toString, (i + 1).toString, (i + 2).toString, (i + 3).toString)
+          })))).toDF("col", "arr")
+          .selectExpr("col", "struct(col, arr) as st")
+          .selectExpr("col", "st.col as col1", "explode(st.arr) as arr_col")
+        df.collect()
+      }
```

Comment:
Please add benchmark result in this PR.

Reply:
This one, `sql/core/benchmarks/MiscBenchmark-results.txt`~

Reply:
Yea, I posted the benchmark result for the added one. Because I think the all benchmark results should be run on r3.xlarge machine, I plan to post it later.

Reply:
Will update `MiscBenchmark-results` soon.